src/Verify/IoHelpers.cs

```diff
@@ -201,8 +202,25 @@ public static string GetRelativePath(string directory, string file)
     private static VirtualizedRunHelper GetForAssembly(Assembly assembly) =>
         virtualizedRunHelpers.GetOrAdd(assembly, a => new(a));
 
-    public static string? GetDirectoryName(string? path) =>
-        Path.GetDirectoryName(GetMappedBuildPath(path));
+    /// <summary>
+    /// Get directory path from a given file path
+    /// </summary>
+    /// <remarks>
+    /// The method accepts input paths mixing / and \ separators
+    /// </remarks>
+    public static string GetDirectoryFromSourceFile(string? sourceFile)
```

Comment:
There are cases where this function is not on par with the `Path.GetDirectoryName` - is that ok?

Reply:
*Sorry for the really short answer, I was on my phone.*

My point is to make it work for what we need and only for that. If it's called `Path.GetDirectoryName` people could think it has the same behavior.

As for the differences with `Path.GetDirectoryName`   I feel like it's not much a problem but I don't really know the lib.

It will be a matter of differences of `/` and `\`. 

The .NET Framework and .NET Core implementation are quite different:
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs,132
https://referencesource.microsoft.com/#mscorlib/system/io/path.cs,156

The .NET Core implementation is really simple, we could copy the implementation and adapt it to support both separators. It would avoid any regression with the current behavior.

Reply:
Humm in fact the Windows implementation has a lot of edge cases: [Windows](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/IO/PathInternal.Windows.cs) vs [Unix](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/IO/PathInternal.Unix.cs)

Reply:
I'm slightli split - probably the best solution would be to mark it `internal` (if possible) - so that it's really used in that specific case - for which the implementation is good enough.

But I do not want to be nitpicking :-)

Overall it looks good and if it solves the case of non-docker, non-WSL cross platform run - then it has it's value and I'm good with it.

Reply:
After thinking more about it I feel like your initial feeling is true:

> but the very first idea is that it'll crash anyway down the road :-/

Is the directory produced will be used to open the verified .txt file and compare the result? If so, we would need to improve `GetMappedBuildPath` to handle non WSL cases.

Reply:
@JanKrivanek I updated `VirtualizedRunHelper` to handle more cases (not just WSL/Docker) and also make it unit test friendly. Let me know what you think.